decred/decred/crypto/rando.py

```diff
@@ -1,17 +1,43 @@
 """
 Copyright (c) 2019, Brian Stafford
-Copyright (c) 2019, The Decred developers
+Copyright (c) 2019-20, The Decred developers
 See LICENSE for details
 """
 
 import os
 
+from decred import DecredError
+from decred.util.encode import ByteArray
+
+
+KEY_SIZE = 32
+HASH_SIZE = 32
 
 MinSeedBytes = 16  # 128 bits
 MaxSeedBytes = 64  # 512 bits
 
 
-def generateSeed(length=MaxSeedBytes):
+def checkSeed(length):
     if length < MinSeedBytes or length > MaxSeedBytes:
-        raise AssertionError("invalid seed length %d" % length)
+        raise DecredError(f"Invalid seed length {length}")
+
+
+def generateSeed(length=MaxSeedBytes):
+    checkSeed(length)
     return os.urandom(length)
+
+
+def newHashRaw():
+    return generateSeed(HASH_SIZE)
+
+
+def newHash():
+    return ByteArray(newHashRaw())
+
+
+def newKeyRaw():
+    return generateSeed(KEY_SIZE)
+
+
+def newKey():
+    return ByteArray(newKeyRaw())
```

Comment:
This does look a bit too abstracted, also given that `HASH_SIZE == KEY_SIZE`, but each of these functions is called at least once in the codebase, and the names increase clarity (assuming correct usage).

Reply:
I'm ok with that, let's see what @buck54321 says.

Reply:
I mean, I think it's fine how it is, but I'm a fan of verbosity.

Reply:
It's not a 32-bit number, it's 32 bytes, so the name would be like `rando.bytes32` or something. 

I do like it better with separate functions though.

Reply:
oh yeah, so 256bit